src/test/ui/did_you_mean/issue-49746-unicode-confusable-in-float-literal-expt.stderr

```diff
@@ -14,5 +14,14 @@ help: Unicode character '−' (Minus Sign) looks like '-' (Minus/Hyphen), but it
 LL | const UNIVERSAL_GRAVITATIONAL_CONSTANT: f64 = 6.674e-11; // m³⋅kg⁻¹⋅s⁻²
    |                                                     ^
 
-error: aborting due to 2 previous errors
+error[E0277]: cannot subtract `{integer}` from `{float}`
```

Comment:
Actually, I think we should do the substitution earlier, so it works in literals too, not just full tokens.

Reply:
The Unicode stuff is legal inside string literals, so I'm not sure there's a *single* earlier point where this can be done.

Reply:
I think that's something I want to do (precisely for string literals) _if and only if_ there is one unambiguous way of parsing it (like somebody copy-pasted code from a PDF and ends up with `println!(‟Hello, world!”);`, but doing so would require a more aggressive implementation with backtracking written in such a way that it doesn't affect the common case, and I don't see it would ever handling anything more complex than `println!(‟It’s 💖”);`. All of that I think _should_ be done, but I'd want it to be part of a separate PR.